src/DynamicDiscreteSamplers.jl

```diff
@@ -606,15 +605,15 @@ SemiResizableWeights(len::Integer) = SemiResizableWeights(FixedSizeWeights(len))
 function FixedSizeWeights(len::Integer)
     m = Memory{UInt64}(undef, allocated_memory(len))
     # m .= 0 # This is here so that a sparse rendering for debugging is easier TODO for tests: set this to 0xdeadbeefdeadbeed
-    m[4:10491+2len] .= 0 # metadata and edit map need to be zeroed but the bulk does not
+    m[4:10491+len] .= 0 # metadata and edit map need to be zeroed but the bulk does not
     m[1] = len
     m[2] = 2051
     # no need to set m[3]
-    m[10235] = 10492+2len
+    m[10235] = 10492+len
     _FixedSizeWeights(m)
 end
-allocated_memory(length::Integer) = 10491 + 8*length # TODO for perf: consider giving some extra constant factor allocation to avoid repeated compaction at small sizes
-length_from_memory(allocated_memory::Integer) = (allocated_memory-10491) >> 3 # Int((allocated_memory-10491)/8)
+allocated_memory(length::Integer) = 10491 + 7*length # TODO for perf: consider giving some extra constant factor allocation to avoid repeated compaction at small sizes
+length_from_memory(allocated_memory::Integer) = Int((allocated_memory-10491)/7)
```

Comment:
This is much slower, but it is only called in `compact!` which is `O(n)` so that's okay

Reply:
yeah, I don't think there is a better way